src/main/java/killrvideo/configuration/DseConfiguration.java

```diff
@@ -129,9 +145,55 @@ private void populateContactPoints(Builder clusterConfig)  {
                     .collect(Collectors.toList());            
         clusterConfig.withPort(clusterNodeAdresses.get(0).getPort());
         clusterNodeAdresses.stream()
-                           .map(adress -> adress.getHostName())
+                           .map(address -> address.getHostName())
                            .forEach(clusterConfig::addContactPoint);
     }
+
+    /**
+     * If SSL is enabled use the supplied CA cert file to create
+     * an SSL context and use to configure our cluster.
+     *
+     * @param clusterConfig
+     *      current configuration
+     */
+    private void populateSSL(Builder clusterConfig) {
```

Comment:
So, just asking - it looks like if someone deploying this has configured SSL to be on, but misconfigured the certificate file in some way, that this catches the error and keeps going, meaning that the app will keep running without SSL. Is this the best behavior? I guess if the server is configured to require SSL, any calls from this client will fail?

Reply:
Right, at this point you won't be able to execute any commands against the database. Yea, I guess we could simply exit out at this point.

Reply:
Exiting with something like IllegalArgument, this is what i would to 
```java
} catch (FileNotFoundException fne) {
  String errMsg = "SSL cert file not found. You must provide a valid certification file when using SSL encryption option.";
  LOGGER.error(errMsg, fne);
  throw new IllegalArgumentException(errMsg, fne);

} catch (CertificateException ce) {
   String errCert = "Your CA certificate looks invalid. You must provide a valid certification file when using SSL encryption option.";
  LOGGER.error(errCert, ce);
  throw new IllegalArgumentException(errCert, ce);

} catch (Exception e) {
  String errSsl = "Exception in SSL configuration: ";
  LOGGER.error(errSsl, e);
  throw new IllegalArgumentException(errSsl, e);
}
```

Reply:
Ok, I pushed some new changes that cleans things up.  I tested the following cases:
no cert file
cert file header broken
cert file footer broken
cert file key incorrect
correct cert file in place

The only one that threw an IllegalArgumentException was when I broke the key.  The others threw the specific exceptions I am explicitly handling for. Stack traces were also properly passed down through the exitOnError() method so they don't get swallowed. I would rather not throw an IllegalArgumentException for things that aren't actually that exception and I think we should definitely exit the application if any of these occur at least so it doesn't look like the app is still running when without the correct cert in place it can't do anything anyway.

Reply:
Oh you are right, no big change. I am not used to `System.exit` and here are the reaon :

- `System.exit` may be dangerous as it is a violent kill, it doesn't even try to go to `finally` block if exists.  Here it is not a big deal but elsewhere in the code you want to execute some `dseSession.getCluster().close()` at shutdown because the driver is stateful and let sockets opened.

- Here both solutions will quit the application immediately. (As the bean definition is in a `@Configuration` class, if you cannot instance the DseSession object you will quit immediately). If you throw an exception you got the stacktrace, the line the error occured and a trace in System.err and System.out. As you explicitly LOG an error when the logger and provides the parent exception it the same i guess.

Reply:
You caught a `checked` Exception but throw a `RuntimeException` (unchecked) to let propagate and stop the application. To my point of view checked exception should be dedicated to business errors - only my feeelings, not a proof whatsoever

Reply:
You...are absolutely correct on leaving sockets opened.  I think I can handle that in my exit method before I call System.exit.  Looking into it.

Reply:
Ooooooook, @clun, so, there are multiple scenarios.  I went and added a @PreDestroy destroy() method to handle the case of the application being shutdown normally.  In the case I explicitly close the cluster connection per your comments.  I also checked this against the DSE documentation to ensure closing the cluster as compared to the session was the way to go.  It is from what I read.  When closing the cluster it will handle the session as well.  So, there is that.

Now on to the other part and how this all started. For the exceptions we are referencing above I cannot actually close the cluster connection because at the point of failure it has not been established yet.  When I tested this and pointed code to destroy() I would get NullPointerExceptions when trying to access the cluster object to close it, again, because it has not been established yet.  With that, I feel we are safe as far as System.exit() goes.

All in all I effectively have 2 "exit" methods.  I'll admit, that seems strange, but they are for different scenarios. Now that I think of it I guess I could test to see if the cluster connection is established in destroy and choose which path from there.  Thoughts?

Reply:
Looks good to me.